### PR TITLE
Feature asserts

### PR DESCRIPTION
**Pull request by Raymundogs**

### CODE REVIEW

File: src/main/java/salesforce/ui/pages/CasesFormPage.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+package salesforce.ui.pages;
+
+import core.selenium.WebElementAction;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class CasesFormPage extends BasePage {
+    private WebElementAction webElementAction = new WebElementAction();
```
Comment (JorgeRCV) [inline src/main/java/salesforce/ui/pages/CasesFormPage.java:18]: I suggest to instaciate this object on the base page to dont set this in every object

File: src/main/java/salesforce/ui/pages/CasesPage.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+package salesforce.ui.pages;
+
+import core.selenium.WebElementAction;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class CasesPage extends BasePage {
+    private WebElementAction webElementAction = new WebElementAction();
```
Comment (JorgeRCV) [inline src/main/java/salesforce/ui/pages/CasesPage.java:17]: Here is another instance

File: src/main/java/salesforce/ui/pages/PopUpConfirmPage.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package salesforce.ui.pages;
+
+import core.selenium.WebElementAction;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class PopUpConfirmPage extends BasePage {
+    private WebElementAction webElementAction = new WebElementAction();
```
Comment (JorgeRCV) [inline src/main/java/salesforce/ui/pages/PopUpConfirmPage.java:9]: Here too

File: src/test/java/salesforce/cases/CreateCasesTests.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        casesFormPage.selectValueOnSlaViolationMenu("No");
+        casesFormPage.inputTextOnSubjectTextBox("The subject");
+        casesFormPage.inputTextOnDescriptionTextBox("The description");
+        casesFormPage.inputTextOnInternalCommentsTextBox("The comments");
+        singleCasePage = casesFormPage.clickOnSaveButton();
+        String actual = singleCasePage.getPopUpMessage();
+        Instant timeStamp = Instant.now();
+        Date expectedDate = DateUtils.truncate(Date.from(timeStamp), Calendar.MINUTE);
+        String expectedStart = "Case";
+        String expectedFinish = "was created.";
```
Comment (JorgeRCV) [inline src/test/java/salesforce/cases/CreateCasesTests.java:111]: Try to use Soft assertions in this test

File: src/main/java/salesforce/ui/pages/CasesFormPage.java [hunk 5]
```diff
@@ -21,8 +21,42 @@
     private WebElement caseOriginMenu;
     @FindBy(xpath = "//span[text()='Priority']/../..//a[@class='select']")
     private WebElement priorityMenu;
+    @FindBy(xpath = "//span[text()='Type']/../..//a[@class='select']")
```
Comment (SaulCM) [inline src/main/java/salesforce/ui/pages/CasesFormPage.java:25]: Web elements with the same structure could be simplified to a single general String xpath and get the element from a method.

**Review by SaulCM — APPROVED** [review]

File: src/test/java/salesforce/cases/CreateCasesTests.java [hunk 6]
```diff
@@ -22,6 +23,13 @@ public void testCreateCaseWithRequiredValues() {
         casesPage = pageTransporter.goToCases();
         casesFormPage = casesPage.clickOnNew();
         casesFormPage.selectValueOnCaseOriginMenu("Phone");
+        singleCasePage = casesFormPage.clickOnSaveButton();
+        String actual = singleCasePage.getPopUpMessage();
+        String expectedStart = "Case";
+        String expectedFinish = "was created.";
+        Assert.assertTrue(actual.startsWith(expectedStart) && actual.contains(expectedFinish));
+        popUpConfirmPage = singleCasePage.clickOnDelete();
+        popUpConfirmPage.clickOnDelete();
```
Comment (SaulCM) [inline src/test/java/salesforce/cases/CreateCasesTests.java:81]: Could also assert the new record name or title in this feature list.

**Review by SaulCM — APPROVED** [review]

File: src/test/java/salesforce/cases/CreateCasesTests.java [hunk 7]
```diff
@@ -25,15 +32,20 @@ public void testCreateCaseWithRequiredValues() {
...
         String actual = singleCasePage.getPopUpMessage();
+        Instant timeStamp = Instant.now();
+        Date expectedDate = DateUtils.truncate(Date.from(timeStamp), Calendar.MINUTE);
         String expectedStart = "Case";
         String expectedFinish = "was created.";
         Assert.assertTrue(actual.startsWith(expectedStart) && actual.contains(expectedFinish));
+        String createdDate = singleCasePage.getCreatedDateLabel();
+        Date actualDate = DateUtils.truncate(new SimpleDateFormat("d/M/yyyy HH:mm").parse(createdDate), Calendar.MINUTE);
+        Assert.assertEquals(expectedDate, actualDate);
         popUpConfirmPage = singleCasePage.clickOnDelete();
```
Comment (SaulCM) [inline src/test/java/salesforce/cases/CreateCasesTests.java:80]: The deletion could be moved to the base test for the group create.

**Review by SaulCM — APPROVED** [review]

**Pull request merged**